runtime/test/package.go

```diff
@@ -0,0 +1,13 @@
+// Package test provides a number of functions and tools for writing fully integrated test suites for Encore
+// applications.
+package test
+
+import (
+	"encore.dev/runtime/config"
+)
+
+func init() {
+	if !config.Cfg.Static.Testing {
+		panic("You can only import the test package from within a test. None of this packages functionality will work outside of a call to `encore test`")
+	}
+}
```

Comment:
Given the intended use for this package is only for testing; and it's going to be deeply tied into the runtime's test support - I think this is a good panic, as it prevents people trying to use functionality provided in this package in their running apps.

Reply:
It's really frustrating to debug these issues, from experience. I think if we want to prevent this we should do it at compile time by inspecting the imports in the parser step, so we can point at where the problem is

Reply:
Fair. Will change to a compiler error if we see an import for this outside test packages